### PR TITLE
x230t : add hotp/non-hotp variant and specialized coreboot config, add x230t to BOARD_TESTERS.md

### LINKED ISSUE #1744: Report: The x230-maximized image is working now on x230t without the backspace keyboard error. But Power button misbehave and no sound card. Seperate board configs needed for x230t
@tlaurion wrote many times and took this even in the year 2024 often as a example for how missing testing is blocking adding the support for a device. The example: x230t. This was famous for not having a working backspace key when a heads x230 image was installed on it. I can report from years ago, that this was the case. The keyboard was not working as expected but back then, i had not the chance to test further/again.
Now i have again access to a x230t and the first thing i did was to install https://output.circle-artifacts.com/output/job/ddd46474-d44a-47ed-8035-d22d97171c0d/artifacts/0/build/x86/x230-maximized/heads-x230-maximized-v0.2.0-2254-g27d09d4.rom
I was expecting it to have the same keyboard issue like years ago, but keyboard worked perfectly fine. Of course at the moment you have to force flash with flashrom because flashrom correctly reports, that the coreboot model number inside the file is not x230t but its x230 (there was a x230t coreboot image on it).

**Is your feature request related to a problem? Please describe.**

There was a problem that seem to be gone on the latest releases.

**Describe the solution you'd like**

Add x230t with default coreboot x230t config and x230t name to heads to finish up the work that started years ago for this model.

**Describe alternatives you've considered**
None?

### PR DESCRIPTION
**Pull request by tlaurion**
Might close  #1744 

---

- x230t do not suffer from backspace and pipe keyboard issues since coreboot 24.02.01 + current 5.10.5 kernel used in master
- x230t soundcard doesn't work
- power button doesn't work

If this PR fixes x230t issues preventing x230t to be used by flashing x230 roms, cool.

Otherwise forget about x230t or deal with upstream coreboot and never talk about x230t or keyboard issues on old Thinpads tablet models until fixed upstream. Wil I do the same for x220t and other tablets? No. The trail below should suffice to any non-developer to create board variants. It's not coding, its porting: this is about plugging things together, not even understanding how things work under the hood. Nonetheless to say: more testers are needed for such boards in case something technical needs to be investigated: I won't have the time to do it if it doesn't work.

For more details, see all linked issues to #1744 
---

Repro for anyone wanting to create a specialized board config pointing to a specialized coreboot/linux config file for a board variant. x230t is x230 but pointing to a different coreboot config file. This is all that is different, se file changes of this PR:
```
#do branch for commit
git checkout -b x230t_addition
#copy/move x230 boards and rename to x230t
cp -r boards/x230-maximized/ boards/x230t-maximized cp -r boards/x230-hotp-maximized boards/x230t-hotp-maximized mv boards/x230t-hotp-maximized/x230-hotp-maximized.config boards/x230t-hotp-maximized/x230t-hotp-maximized.config mv boards/x230t-maximized/x230-maximized.config boards/x230t-maximized/x230t-maximized.config cp config/coreboot-x230-maximized.config config/coreboot-x230t-maximized.config vim boards/x230t-maximized/x230t-maximized.config
#point boards to x230t to be modified coreboot config file
vim boards/x230t-hotp-maximized/x230t-hotp-maximized.config 
# call Heads coreboot Makefile helper to modify coreboot config with menuconfig in place
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make BOARD=x230t-maximized coreboot.modify_and_save_oldconfig_in_place 
#Add board configs and coreboot config changes in to be git commit
git add boards/x230t-hotp-maximized/x230t-hotp-maximized.config boards/x230t-maximized/x230t-maximized.config config/coreboot-x230t-maximized.config 
#add x230t board names to be built by CircleCI by this commit
vim .circleci/config.yml
git add .circleci/config.yml
#Add @fhvyhjriur to x230t board testers in tracking file
vim BOARD_TESTERS.md
git add BOARD_TESTERS.md
#create git commit, signing commit with message
git commit --signoff -m
#push this commit to a PR.
git push --force tlaurion-github
```

@fhvyhjriur please test and report...

### CODE REVIEW

**Comment by tlaurion** [comment]
@fhvyhjriur another bit rotting PR without testing what you asked for. :(

**Comment by kieroid** [comment]
testing with my x230t. i will eat dirt until it's tested and merged.

**Comment by kieroid** [comment]
do you mind re-running the circleCI job in regards to this pull request? the artifacts that were produced are now already expired. 😎👍

**Comment by tlaurion** [comment]
> > testing with my x230t. i will eat dirt until it's tested and merged.
> > @kieroid not sure what this expression means in this context (not english native speaker, it seems you mean that you need it and wait for it?)
> 
> This PR is really old. Can try to rebase on master?

Here you go @kieroid. check for artifacts under greenmark for x230t-hotp-maximized/x230t-maximized board builds for top bottom roms to be externally flashed, ask questions here if you have any relevant to this ticket otherwise refer to documentation at https://osresearch.net or go under community page to discuss things live if needed. 

Roms should be built within next hour or so and downloadable from documentated ways.

**Comment by kieroid** [comment]
thank you sir

**Comment by kieroid** [comment]
it seems that the kernel (and possibly the x230t's board config files) are undefined/don't exist. would it be possible that boards incompatible with heads were previously removed from the config directory?

either way, this doesn't seem like a difficult fix. i may be able to do this. i will get to it when i have time.

File: boards/x230t-hotp-maximized/x230t-hotp-maximized.config [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#export CONFIG_ENABLE_FUNCTION_TRACING_OUTPUT=y
+
+#On-demand hardware support (modules.cpio)
+CONFIG_LINUX_USB=y
+CONFIG_LINUX_E1000E=y
+CONFIG_MOBILE_TETHERING=y
+
+#Modules packed into tools.cpio
+CONFIG_CRYPTSETUP2=y
+CONFIG_FLASHROM=y
```
Comment (kieroid) [inline boards/x230t-hotp-maximized/x230t-hotp-maximized.config:27]: CONFIG_FLASHROM -> CONFIG_FLASHPROG

**Comment by tlaurion** [comment]
Other things moved, this is why we need PR tested without them bitrotting. Plate full. 
Error in CirlceCI build tells:
- kernel version pointed is not existing anymore or wrong
- yes, move to flashprog, check that everything is ~same as x230. It should then work.

**Comment by kieroid** [comment]
i will test it and submit a PR of my own soon enough.
i know it's not really a priority of yours! (it's not really a priority of mine either.)
although, it would be nice to add more (official) support for different boards.

**Comment by kieroid** [comment]
no dice on this PR. changing that line within the coreboot config doesn't change anything (including the backspace bug.)

when the computer is rebooted, everything works perfectly. i think considering this, it might just be worth closing this PR request (as the x230 rom is essentially the exact same as the x230t rom.)

it is what it is.

**Pull request closed without merging**